src/game/savegame_bson.c

```diff
@@ -647,59 +666,79 @@ static bool Savegame_BSON_LoadLara(
-        json_object_append_int(start_obj, "num_medis", start->num_medis);
+            resume_obj, "shotgun_ammo", resume->shotgun_ammo);
+        json_object_append_int(resume_obj, "num_medis", resume->num_medis);
         json_object_append_int(
-            start_obj, "num_big_medis", start->num_big_medis);
-        json_object_append_int(start_obj, "num_scions", start->num_scions);
-        json_object_append_int(start_obj, "gun_status", start->gun_status);
-        json_object_append_int(start_obj, "gun_type", start->gun_type);
-        json_object_append_bool(start_obj, "available", start->flags.available);
+            resume_obj, "num_big_medis", resume->num_big_medis);
+        json_object_append_int(resume_obj, "num_scions", resume->num_scions);
+        json_object_append_int(resume_obj, "gun_status", resume->gun_status);
+        json_object_append_int(resume_obj, "gun_type", resume->gun_type);
         json_object_append_bool(
-            start_obj, "got_pistols", start->flags.got_pistols);
+            resume_obj, "available", resume->flags.available);
         json_object_append_bool(
-            start_obj, "got_magnums", start->flags.got_magnums);
-        json_object_append_bool(start_obj, "got_uzis", start->flags.got_uzis);
+            resume_obj, "got_pistols", resume->flags.got_pistols);
         json_object_append_bool(
-            start_obj, "got_shotgun", start->flags.got_shotgun);
-        json_object_append_bool(start_obj, "costume", start->flags.costume);
-        json_array_append_object(start_arr, start_obj);
+            resume_obj, "got_magnums", resume->flags.got_magnums);
+        json_object_append_bool(resume_obj, "got_uzis", resume->flags.got_uzis);
+        json_object_append_bool(
+            resume_obj, "got_shotgun", resume->flags.got_shotgun);
+        json_object_append_bool(resume_obj, "costume", resume->flags.costume);
+        json_object_append_int(resume_obj, "timer", resume->stats.timer);
+        json_object_append_int(resume_obj, "kills", resume->stats.kill_count);
+        json_object_append_int(
+            resume_obj, "secrets", resume->stats.secret_flags);
+        json_object_append_int(
+            resume_obj, "pickups", resume->stats.pickup_count);
+        json_object_append_int(resume_obj, "deaths", resume->stats.death_count);
+        json_object_append_int(
+            resume_obj, "max_kills", resume->stats.max_kill_count);
+        json_object_append_int(
+            resume_obj, "max_secrets", resume->stats.max_secret_count);
+        json_object_append_int(
+            resume_obj, "max_pickups", resume->stats.max_pickup_count);
+        json_array_append_object(resume_arr, resume_obj);
     }
-    return start_arr;
+    return resume_arr;
 }
 
-static struct json_array_s *Savegame_BSON_DumpEndInfo(GAME_INFO *game_info)
+static struct json_array_s *Savegame_BSON_DumpCurrentInfo(GAME_INFO *game_info)
```

Comment:
There's no need to have a separate function for serializing current info. (Deserializing is only for backwards compatibility.) We should use `Savegame_BSON_DumpResumeInfo` for both start and current.